### PR TITLE
Install-driven enhancements, round 2

### PR DESCRIPTION
**Pull request by ydirson**

### CODE REVIEW

File: lib/host.py
Comment (stormi) [inline lib/host.py:570]: What about moving it to the VDI class?

Comment (ydirson) [reply lib/host.py:570]: My thought was rather to move it to `Pool` instead.  In `VDI` it would need to be a class- or static-method, and would need to get a ref to the pool anyway.

File: lib/pool.py
Comment (stormi) [inline lib/pool.py:26]: This could be useful in other places, if this implementation is better than the existing one which does a similar check by waiting for `xe` to return that the current host is enabled, as we do each time we reboot a host or restart XAPI.

I don't know which is safer:
* I'm not 100% confident in this `xapi-wait-init-complete` check
* I've seen the enabled status displayed by XO flip to enable, then shortly to disabled again, then enabled again for good.

I'd start with making it a helper method of the Host class, and decide later in what direction to harmonize (maybe both checks added together?)

File: lib/vm.py
Comment (stormi) [inline lib/vm.py:479]: I'd rather not merge this change until a better handling is done, otherwise cloned VMs created from cached VMs will be hard to differentiate from cache by their name.

File: lib/host.py
Comment (stormi) [reply lib/host.py:570]: To me, the SR looks like an attribute of the VDI object. Why would it be a static method? You even added the `VDI.param_get` method that you could use.

File: lib/vm.py
Comment (ydirson) [reply lib/vm.py:479]: yes there is a FIXME for this already

Comment (stormi) [reply lib/vm.py:479]: I didn't know if the FIXME was meant for before or after the merge :)

Comment (ydirson) [reply lib/vm.py:479]: inside commit message is explicitly *before* ;)

File: lib/host.py
Comment (ydirson) [reply lib/host.py]: The caller has no VDI object to start with, we want to get one from just a UUID.

Comment (ydirson) [reply lib/host.py]: (moved to `Pool` in last push)

File: lib/vm.py
Comment (ydirson) [reply lib/vm.py]: done in last push

**Review by ydirson — COMMENTED** [review]
Note this includes the use of `rt_answerfile` as workaround to the Restore

**Review by gthvn1 — APPROVED** [review]

File: lib/host.py
Comment (stormi) [reply lib/host.py]: I thought we could instanciate objets, but there's a chicken and egg problem, since we ask for a SR object in the constructor. For now, a `Pool` method will do.

**Comment by ydirson** [comment]
Reworked/cleaned the `install_xcpng` split into `pxe.py`, and reinstated the `rt_answerfile` usage.
Brings this PR out of Draft.

File: lib/pool.py
Comment (ydirson) [reply lib/pool.py:26]: > This could be useful in other places

In theory yes, but in practice, how often do we interact with XAPI without building a `Pool` object?

> I'm not 100% confident in this xapi-wait-init-complete check

Neither do I, but I have not seen any practical issue with it, and it does fix those flakiness issues AFAICT.

> I've seen the enabled status displayed by XO flip to enable, then shortly to disabled again, then enabled again for good.

That would look like a bug in XAPI

**Review by stormi — APPROVED** [review]

**Review by gduperrey — APPROVED** [review]
Run CI on 8.3 :white_check_mark:

**Review by gthvn1 — APPROVED** [review]

**Pull request merged**
